pkg/informer/informer.go

```diff
@@ -44,6 +75,160 @@ func InformerForResource(res schema.GroupVersionResource) (GenericInformer, erro
+					if rec == kind {
+						klog.V(1).Infof("Resource %s %s matched deny list. ", group, kind)
+
+						return false
+					} else {
+						if api != "*" && rec == "*" {
+							if group == api {
+								klog.V(1).Infof("Resource %s %s matched deny list. ", group, kind)
+								return false
+							}
+						}
+
+					}
+				} else {
+					// if api != "*" && deny.Resources[i] != "*" {
+					if group == api && kind == rec {
+						klog.V(1).Infof("Resource %s %s matched deny list. ", group, kind)
+						return false
+					}
+					// }
+				}
+			}
+		}
+	}
+
+	// if allowedlist is empty allow all resources, otherwise if * allow all groups specific
+	// resources if not * allow specific resources to specific groups:
+	if len(allowedList) == 0 {
+		return true
+	} else {
+		for _, allow := range allowedList {
+			for _, api := range allow.ApiGroups {
+				for _, rec := range allow.Resources {
+					if api == "*" && rec != "*" { //all apigroups & specific resources
+						if rec == kind {
+							klog.V(1).Infof("Resource %s %s matched allow list. ", group, kind)
+							return true
+
+						} else {
+							if api != "*" && rec == "*" { // specific apigroups & all resources
+								if group == api {
+									klog.V(1).Infof("Resource %s %s matched allow list. ", group, kind)
+									return true
+								}
+							}
+						}
+
+					} else {
+						// if api != "*" && allow.Resources[i] != "*" { //specific apigroups and resources
+						if group == api && kind == rec {
+							klog.V(1).Infof("Resource %s %s matched allow list. ", group, kind)
+							return true
+						}
+						// }
+					}
+				}
+			}
+		}
+	}
+
+	klog.Warningf("Resource %s %s missing case.", kind, group)
+
+	return true
+}
+
+// Returns a map containing all the GVRs on the cluster of resources that support WATCH (ignoring clusters and events).
+func SupportedResources(discoveryClient *discovery.DiscoveryClient) (map[schema.GroupVersionResource]struct{}, error) {
+	// Next step is to discover all the gettable resource types that the kuberenetes api server knows about.
+	supportedResources := []*machineryV1.APIResourceList{}
+
+	// List out all the preferred api-resources of this server.
+	apiResources, err := discoveryClient.ServerPreferredResources() //<--here we can look into this list and have preferred versions
+	if err != nil && apiResources == nil {                          // only return if the list is empty
+		return nil, err
+	} else if err != nil {
+		glog.Warning("ServerPreferredResources could not list all available resources: ", err)
+	}
+
+	// create client to get configmap
+	config := config.GetKubeConfig() //can't err here?
+	clientset, err := kubernetes.NewForConfig(config)
+
+	//locate the allow-deny ConfigMap:
+	var cm *v1.ConfigMap
+	if cm, _ = clientset.CoreV1().ConfigMaps("POD_NAMESPACE").Get(contextVar, "allowdeny-config", metav1.GetOptions{}); err != nil {
+		glog.Warning("Can't find allowdeny-config ConfigMap", err)
```

Comment:
Not a warning because this is optional config, but lets make this very descriptive to help end users as much as possible.
```
glog.Info("Didn't find ConfigMap with name allowdeny-config. Will collect all resources. ", err)
```